polus-graph-pyramid-builder-plugin/src/main.py

```diff
@@ -190,138 +489,294 @@ def format_ticks(fmin,fmax,nticks):
+                if i < 0:
+                    formtick = str(decformtick[:5]) + newprefix
+                else: 
+                    formtick = str(decformtick[:4]) + newprefix
+        elif convertexponent < 0:
+            if convertexponent % -3 == -2:
+                movednum = round(numbers*10,1)
+                newprefix = _prefix[int(convertexponent - 1)]
+                formtick = str(movednum) + newprefix
+            elif convertexponent % -3 == -1:
+                movednum = round(numbers/10,2)
+                newprefix = _prefix[int(convertexponent + 1)]
+                formtick = str(movednum) + newprefix
+            else:
+                newprefix = _prefix[convertexponent]
+                if i < 0:
+                    formtick = str(decformtick[:5]) + newprefix
+                else: 
+                    formtick = str(decformtick[:4]) + newprefix
+        else:
+            if i < 0:
+                formtick = str(decformtick[:5]) + _prefix[int(convertexponent)]
+            else: 
+                formtick = str(decformtick[:4]) + _prefix[int(convertexponent)]
+        convertprefix.append(int(convertexponent))
+        fticks.append(formtick)
 
-def get_cmap():
-    # Create a custom colormap to mimick Polus Plots
+    return fticks
 
-    cmap_values = [[1.0,1.0,1.0,1.0]]
-    cmap_values.extend([[r/255,g/255,b/255,1] for r,g,b in zip(np.arange(0,255,2),
+# Create a custom colormap to mimick Polus Plots
+def get_cmap(type):
+    
+    if type == "linear":
+        cmap_values = [[1.0,1.0,1.0,1.0]]
+        cmap_values.extend([[r/255,g/255,b/255,1] for r,g,b in zip(np.arange(0,255,2),
                                                             np.arange(153,255+1/128,102/126),
                                                             np.arange(34+1/128,0,-34/126))])
-    cmap_values.extend([[r/255,g/255,b/255,1] for r,g,b in zip(np.arange(255,136-1/128,-119/127),
+        cmap_values.extend([[r/255,g/255,b/255,1] for r,g,b in zip(np.arange(255,136-1/128,-119/127),
                                                             np.arange(255,0,-2),
                                                             np.arange(0,68+1/128,68/127))])
-    cmap = ListedColormap(cmap_values)
-
+        cmap = ListedColormap(cmap_values)
+    if type == "log":
+        cmap_values = [[1.0,1.0,1.0,1.0]]
+        cmap_values.extend([[r/255,g/255,b/255,1] for r,g,b in zip(np.arange(0,255,2),
+                                                            np.arange(153,255+1/128,102/126),
+                                                            np.arange(34+1/128,0,-34/126))])
+        cmap_values.extend([[r/255,g/255,b/255,1] for r,g,b in zip(np.arange(255,136-1/128,-119/127),
+                                                            np.arange(255,0,-2),
+                                                            np.arange(0,68+1/128,68/127))])
+        cmap = ListedColormap(cmap_values)
```

Comment:
I don't see the need to have an if/else statement here. Isn't code in each block the same?

Reply:
Not sure why I did that